### ISSUE TITLE
How to adjust hyper parameters for horovod multi-gpu training?

### ISSUE THREAD

**Issue by y0un5** [google/automl#468]
When use horovod for multi-gpu training, should I manually adjust hyper parameters such as train_batch_size, num_examples_per_epoch, num_epochs? Or they are adjusted automatically?

For example, there is a sample command in README. If I change `num_gpus` from 1 to 2, 4, or 8 ... , what are proper values for train_batch_size, num_examples_per_epoch, num_epochs?
```
!horovodrun -np <num_gpus> -H localhost:<num_gpus> python main.py --mode=train \
    --training_file_pattern=tfrecord/pascal*.tfrecord \
    --validation_file_pattern=tfrecord/pascal*.tfrecord \
    --model_name=efficientdet-d0 \
    --model_dir=/tmp/efficientdet-d0-finetune  \
    --ckpt=efficientdet-d0  \
    --train_batch_size=8 \
    --eval_batch_size=8 --eval_samples=1024 \
    --num_examples_per_epoch=5717 --num_epochs=1  \
    --hparams=voc_config.yaml
    --strategy=horovod
```

**Comment by mingxingtan** [comment]
+ @fsx950223 

Learning rate is automatically adjusted: https://github.com/google/automl/blob/11574fd1c1aaf1ca3dc37639aa365802ba21f099/efficientdet/det_model_fn.py#L517

I believe train batch size is also automatically adjusted, but you need to set num_examples_per_epoch, num_epochs by yourself (since they are specific to datasets).

**Comment by fsx950223** [comment]
It's adjusted by horovod

**Comment by y0un5** [comment]
@mingxingtan @fsx950223

Thanks for the answers! I have one more question about total batch size.

Suppose I have the following settings for 1 GPU.

```
horovodrun -np 1 -H localhost:1 python main.py --mode=train \
     ...
    --train_batch_size=8 \
    --learning_rate=0.001 \
     ...
    --strategy=horovod
```

If I change the number of GPU to 2, then the actual learning rate will be 0.002 by `learning_rate = learning_rate * hvd.size()`, right?

Then what is the total batch size? Still 8(4 for each GPU) or 8 * 2 = 16?

**Comment by fsx950223** [comment]
8 * 2 = 16

**Comment by y0un5** [comment]
Ok, got it! So, I use settings for 1 GPU in horovodrun and just change the number of GPU, then horovod will adjust learning rate and total batch size by multiplying the number of GPU. 

Thanks a lot!

**Comment by sadransh** [comment]
@fsx950223 these stay the same with tf.estimator?
